src/js/ripple/keypairs.js

```diff
@@ -0,0 +1,283 @@
+'use strict';
+
+/* -------------------------------- REQUIRES -------------------------------- */
+
+var util = require('util');
+var nacl = require('tweetnacl');
+var utils = require('./utils');
+var Crypt = require('./crypt').Crypt;
+var Seed = require('./seed').Seed;
+var UInt160 = require('./uint160').UInt160;
+
+var arrayToHex = utils.arrayToHex;
+var hexToArray = utils.hexToArray;
+var sjcl = utils.sjcl;
+
+/* ---------------------------------- ENUMS --------------------------------- */
+
+var KeyType = exports.KeyType = {
+  secp256k1: 'secp256k1',
+  ed25519: 'ed25519'
+};
+
+/* ----------------------------------- OO ----------------------------------- */
+
+function isVirtual() {
+  throw new Error('virtual method not implemented ');
+}
+
+function hasCachedProperty(obj, name, computer) {
+  var key = name + '__';
+  obj[name] = function() {
+    return this[key] !== undefined ? this[key] :
+           this[key] = computer.call(this);
+  };
+}
+
+/* --------------------------------- HELPERS -------------------------------- */
+
+function toGenericArray(typedArray) {
+  var generic = [];
+  Array.prototype.push.apply(generic, typedArray);
+  return generic;
+}
+
+/*
+@private
+@param {Object} opts
+@param {String} [opts.seedBase58] -
+@param {String} [opts.seedHex] -
+@param {String} [opts.passphrase] -
+@param {String} [opts.generic] -
+ */
+function parseSeed(opts) {
+  var seed = new Seed();
+
+  if (opts.passphrase) {
+    seed.parse_passphrase(opts.passphrase);
+  } else if (opts.seedBase58) {
+    seed.parse_base58(opts.seedBase58);
+  } else if (opts.seedHex) {
+    seed.parse_hex(opts.seedHex);
+  } else if (opts.generic) {
+    seed.parse_json(opts.generic);
+  } else {
+    throw new Error('generic, seed, seed_hex, or passphrase must be supplied');
+  }
+
+  return seed;
+}
+
+/*
+@private
+@param {Seed} seed
+ */
+function deriveEdKeyPairSeed(seed) {
+  var hashed = sjcl.hash.sha512.hash(seed.to_bits());
+  var bits = sjcl.bitArray.bitSlice(hashed, 0, 256);
+  return new Uint8Array(sjcl.codec.bytes.fromBits(bits));
+}
+
+function bytesToHex(bytes) {
+  return arrayToHex(bytes).toUpperCase();
+}
+
+function bytesBnTo256Bits(bytes) {
+  var bits = sjcl.codec.bytes.toBits(bytes);
+  var bitLength = 256;
+  return sjcl.bn.fromBits(bits).toBits(bitLength);
+}
+
+function createAccountId(publicKeyBytes) {
+  var bits = sjcl.codec.bytes.toBits(publicKeyBytes);
+  var hash = sjcl.hash.ripemd160.hash(sjcl.hash.sha256.hash(bits));
+  var id = UInt160.from_bits(hash);
+  id.set_version(0);
+  return id;
+}
+
+function stringifyKey(specifier) {
+  if (typeof specifier === 'object') {
+    var sorted = {};
+    Object.keys(specifier).forEach(function(k) {
```

Comment:
This was discussed elsewhere. The order of object keys is never guaranteed but I think this will work most of the time as-is